Marketplace/MarketplaceToolkit.ps1

```diff
@@ -0,0 +1,2109 @@
+    
+            if (($dimensions.width -ne $image.width) -or ($dimensions.height -ne $image.height)){ 
+                $Script:validation_error = $true
+                $message = ($message + " The selected image is " + $dimensions.width + "x" + $dimensions.height) 
+                }
+            }
+        }
+    }
+if ($vmextension){
+    if ($Script:PackageType -eq 'Extension'){
+        $ContentJSON = Get-Content ($field_value)
+        $params = ($ContentJSON | ConvertFrom-Json).parameters 
+        $params = $params.psobject.members | where {$_.membertype -eq 'NoteProperty'}
+        if (($params | where {(($_.name -eq 'vmName') -and ($_.value.type -eq 'string')) -or (($_.name -eq 'location') -and ($_.value.type -eq 'string'))}).count -lt 2){ 
+        $Script:validation_error = $true 
+        $message = 'Required string parameters "vmName" and "location" not found in template'
+        }
+}
+}
+if ($nocondition){ 
+        $Script:validation_error = $true 
+}
+if ($empty){
+    if ($field_value.length -eq 0) { 
+        $message = 'Required field'
+        $Script:validation_error = $true 
+    }
+}
+
+# Validation Actions
+$control = Get-Variable $field
+
+if ($Script:validation_error) {
+        #parent stackpanel
+        $tooltip = new-object System.Windows.Controls.ToolTip
+        $tooltip.Background = "#282D32"
+        $tooltip.Foreground = "white"
+        $tooltip.BorderThickness = 0
+        $tooltip.Padding = 15
+        $tooltip.HorizontalOffset = -5
+        $tooltip.VerticalOffset = -50
+        $tooltip.Content = $message
+        $tooltip.Placement = "left"
+        $control.value.parent.ToolTip = $tooltip
+        #child textbox
+        $control.value.Width = ([int]$columnwidth - [int]15)
+        $control.value.BorderBrush="Red"
+        #child button
+        $button = $control.value.parent.children | where {$_.GetType().fullname -eq 'System.Windows.Controls.Button' -and $_.Content -eq '!'} 
+        $button.visibility = 'Visible'
+        }
+ else {
+        #parent stackpanel
+        $control.value.parent.ToolTip = $null
+        #child textbox
+        $control.value.Width = $columnwidth
+        $control.value.BorderBrush="#FFABADB3"
+        #child button
+        $button = $control.value.parent.children | where {$_.GetType().fullname -eq 'System.Windows.Controls.Button' -and $_.Content -eq '!'} 
+        $button.visibility = 'Collapsed'
+        }
+}
+
+function F_Browse {
+Param(
+[string]$title,
+[string]$filter
+)
+[System.Reflection.Assembly]::LoadWithPartialName("System.Windows.Forms") | Out-Null
+	$Script:F_Browse_obj = New-Object System.Windows.Forms.OpenFileDialog
+	$Script:F_Browse_obj.Filter = $filter
+	$Script:F_Browse_obj.Title = $title
+	$Script:F_Browse_obj.ShowDialog()
+}
+
+function F_Get_ParamFile {
+Param(
+  [string]$File
+)
+Import-Module $File -Force
+if ($applicationName) { $X_Input_Tbx_Name.Text = $applicationName }
+if ($publisher) { $X_Input_Tbx_Publisher.Tex
```

Comment:
args[X] here is hard to read and maintain 
Do you think use one single argument as hashtable can be better? you then assign to local var using $args[0]["Endpoint"]